r/R/csv.R

```diff
@@ -718,6 +724,9 @@ write_csv_arrow <- function(x,
     csv___WriteCSV__RecordBatch(x, write_options, sink)
   } else if (inherits(x, "Table")) {
     csv___WriteCSV__Table(x, write_options, sink)
+  } else if (inherits(x, "Dataset")) {
```

Comment:
```suggestion
  } else if (inherits(x, c("Dataset", "arrow_dplyr_query"))) {
```

and fix the checks above. Scanner$create() will handle both. There's no reason to pull a query into an R data.frame, is there?